File: test/ayush.dev/python/pipelines/PythonInsideProjectDatasetsOnly/code/pythoninsideprojectdatasetsonly/graph/Aggregate_1.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from pythoninsideprojectdatasetsonly.config.ConfigStore import *
from pythoninsideprojectdatasetsonly.udfs.UDFs import *

def Aggregate_1(spark: SparkSession, in0: DataFrame) -> DataFrame:
    df1 = in0.groupBy(col("last_name"), col("firstname"))
    df2 = df1.pivot("c_concat", ["customer_id"])

    return df2.agg(
        first(col("account_flags")).alias("account_flags"), 
        first(col("account_open_date")).alias("account_open_date")
    )

File: ayush.dev/abhisheks+shared/GPDScGit_CloneExternal/pipelines/ScalaInsideProjectDatasetsOnly/code/src/main/scala/io/prophecy/pipelines/scalainsideprojectdatasetsonly/graph/Aggregate_1.scala
package io.prophecy.pipelines.scalainsideprojectdatasetsonly.graph

import io.prophecy.libs._
import io.prophecy.pipelines.scalainsideprojectdatasetsonly.config.ConfigStore._
import io.prophecy.pipelines.scalainsideprojectdatasetsonly.udfs.UDFs._
import io.prophecy.pipelines.scalainsideprojectdatasetsonly.udfs._
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import org.apache.spark.sql.expressions._
import java.time._

object Aggregate_1 {

  def apply(spark: SparkSession, in: DataFrame): DataFrame =
    in.groupBy(col("c_decimal"), col("c_float"), col("c_boolean"))
      .pivot(col("c_string"),    List("c_timestamp", "c_date"))
      .agg(first(col("c_short")).as("c_short"),
           first(col("c_long")).as("c_long")
      )

}
